src/modm/platform/clock/sam_pmc/clockgen_impl.hpp.in

```diff
@@ -11,34 +11,89 @@
 
 namespace modm::platform {
 
-template <MasterClkSource src, MasterClkPrescaler pres>
+template <MainInternalFreq freq>
+void ClockGen::enableMainInternal() {
+	// Enable Main RC Oscillator and wait for it to stabilize
+	PMC->CKGR_MOR |= CKGR_MOR_KEY_PASSWD | CKGR_MOR_MOSCRCEN;
+	while (!(PMC->PMC_SR & PMC_SR_MOSCRCS));
+
+	// Change Main RC Oscillator frequency
+	PMC->CKGR_MOR =
+		(PMC->CKGR_MOR & ~(CKGR_MOR_MOSCRCF_Msk)) |
+		CKGR_MOR_MOSCRCF((uint32_t)freq) |
+		CKGR_MOR_KEY_PASSWD;
+	while (!(PMC->PMC_SR & PMC_SR_MOSCRCS));
+}
+
+template <MainClockSource src>
```

Comment:
Same here, the branching code for a bool is very efficient on Cortex-M, there no need to optimize this.